2025-06-13 02:50:30 - ERROR - Error in update_game: 'str' object has no attribute 'unlocked' | Info: turn=1
Traceback (most recent call last):
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 412, in update_game
    'unlocked_technologies': [tech.name for tech in self.technology_tree.technologies if tech.unlocked]
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 412, in <listcomp>
    'unlocked_technologies': [tech.name for tech in self.technology_tree.technologies if tech.unlocked]
                                                                                         ^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'unlocked'
2025-06-13 02:50:45 - ERROR - Error in update_game: 'str' object has no attribute 'unlocked' | Info: turn=2
Traceback (most recent call last):
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 412, in update_game
    'unlocked_technologies': [tech.name for tech in self.technology_tree.technologies if tech.unlocked]
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 412, in <listcomp>
    'unlocked_technologies': [tech.name for tech in self.technology_tree.technologies if tech.unlocked]
                                                                                         ^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'unlocked'
2025-06-13 02:51:00 - ERROR - Error in update_game: 'str' object has no attribute 'unlocked' | Info: turn=3
Traceback (most recent call last):
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 412, in update_game
    'unlocked_technologies': [tech.name for tech in self.technology_tree.technologies if tech.unlocked]
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 412, in <listcomp>
    'unlocked_technologies': [tech.name for tech in self.technology_tree.technologies if tech.unlocked]
                                                                                         ^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'unlocked'
2025-06-13 02:51:15 - ERROR - Error in update_game: 'str' object has no attribute 'unlocked' | Info: turn=4
Traceback (most recent call last):
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 412, in update_game
    'unlocked_technologies': [tech.name for tech in self.technology_tree.technologies if tech.unlocked]
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 412, in <listcomp>
    'unlocked_technologies': [tech.name for tech in self.technology_tree.technologies if tech.unlocked]
                                                                                         ^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'unlocked'
2025-06-13 02:51:30 - ERROR - Error in update_game: 'str' object has no attribute 'unlocked' | Info: turn=5
Traceback (most recent call last):
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 412, in update_game
    'unlocked_technologies': [tech.name for tech in self.technology_tree.technologies if tech.unlocked]
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 412, in <listcomp>
    'unlocked_technologies': [tech.name for tech in self.technology_tree.technologies if tech.unlocked]
                                                                                         ^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'unlocked'
2025-06-13 02:53:49 - ERROR - Error in update_game: 'list' object has no attribute 'items' | Info: turn=2
Traceback (most recent call last):
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 414, in update_game
    self.objectives_panel.update_objectives(game_state)
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\gui\objectives_panel.py", line 251, in update_objectives
    self.objective_manager.update_objectives(game_state)
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\core\objectives.py", line 454, in update_objectives
    for building_name, building_data in buildings.items():
                                        ^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'items'
2025-06-13 02:54:04 - ERROR - Error in update_game: 'list' object has no attribute 'items' | Info: turn=3
Traceback (most recent call last):
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\Main.py", line 414, in update_game
    self.objectives_panel.update_objectives(game_state)
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\gui\objectives_panel.py", line 251, in update_objectives
    self.objective_manager.update_objectives(game_state)
  File "C:\Users\TUF\python\symulator_miasta_projekt\City_Builder\core\objectives.py", line 454, in update_objectives
    for building_name, building_data in buildings.items():
                                        ^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'items'
